mne/simulation/source.py

```diff
@@ -265,38 +270,50 @@ def simulate_stc(src, labels, stc_data, tmin, tstep, value_fun=None):
             data = np.outer(values_sel, stc_data[i])
         else:
             data = np.tile(stc_data[i], (len(src_sel), 1))
+        # If overlaps are allowed, deal with them
+        if allow_overlap:
+            # Search for the positions of the duplicate vertex indices
+            # in the existing vertex matrix vertex. The size of ov_ind
+            # is the same as src_sel, and for each element in src_sel,
+            # it contains either the position of that vertex in vertno,
+            # or an empty array if the vertex is not a duplicate
+            ov_ind = [np.squeeze(np.nonzero(v == vertno[hemi_ind]))
+                      for v in src_sel]
+            duplicates = []
+            for ind, src_ind in zip(ov_ind, range(len(src_sel))):
+                # If the dimension is zero, it's a scalar,
+                # If it is empty, ndim returns 1
+                if ind.ndim == 0:
+                    # Add the new data to the existing one
+                    stc_data_extended[hemi_ind][ind] += data[src_ind]
+                    duplicates.append(src_ind)
```

Comment:
I would potentially simplify a bit as:
```
for src_ind in src_sel:
    ind = np.where(src_ind == vertno[hemi_ind])[0]  # guaranteed 1D
    if len(ind) > 0:
        stc_data_extended[hemi_ind][ind] += data[src_ind]
        duplicates.append(src_ind)
```
It loops once instead of twice and maintains 1D data which is nice. (I generally avoid `np.squeeze` because it has non-obvious behaviors sometimes.) Can you see if you like it better and if so, if it's equivalent?

Reply:
It's all good, except for the fact that we want the physical indices of both data and stc_data_extended[] so I still need to replace where with nonzero and enumerate the src_sel items. When I do it like that everything seems to work in exactly the same way.